tests/features/permissions.spec.ts

```diff
@@ -207,216 +207,148 @@ export function testPermissions(): void {
+      const alice = await TestDataGenerator.generateDidKeyPersona();
+      const bob = await TestDataGenerator.generateDidKeyPersona();
 
-        const alice = await TestDataGenerator.generateDidKeyPersona();
-        const bob = await TestDataGenerator.generateDidKeyPersona();
+      const permissionScope = {
+        interface : DwnInterfaceName.Records,
+        method    : DwnMethodName.Write
+      };
 
-        // 1. Verify that a permission request is rejected with the data payload is invalid
-        const invalidPermissionRequestData = {
-          description: 'missing required properties such as `scope`'
-        };
+      const grantWrite = PermissionsProtocol.createGrant({
+        signer      : Jws.createSigner(alice),
+        dateExpires : Time.createOffsetTimestamp({ seconds: 100 }),
+        description : 'Allow Bob to write',
+        grantedTo   : bob.did,
+        scope       : permissionScope as any // explicity as any to test the validation
+      });
+      expect(grantWrite).to.eventually.be.rejectedWith(DwnErrorCode.PermissionsProtocolCreateGrantRecordsScopeMissingProtocol);
+    });
 
-        const requestBytes = Encoder.objectToBytes(invalidPermissionRequestData);
-        const requestRecordsWrite = await RecordsWrite.create({
-          signer       : Jws.createSigner(bob),
-          protocol     : PermissionsProtocol.uri,
-          protocolPath : PermissionsProtocol.requestPath,
-          dataFormat   : 'application/json',
-          data         : requestBytes,
-        });
-        const requestWriteReply = await dwn.processMessage(
-          alice.did,
-          requestRecordsWrite.message,
-          { dataStream: DataStream.fromBytes(requestBytes) }
-        );
-
-        expect(requestWriteReply.status.code).to.equal(400);
-        expect(requestWriteReply.status.detail).to.contain(DwnErrorCode.SchemaValidatorFailure);
-
-        // 2. Verify that a permission grant is rejected with the data payload is invalid
-        const invalidPermissionGrantData = {
-          description: 'missing required properties such as `scope`'
+    it('should fail if an invalid protocolPath is used during Permissions schema validation', async () => {
+      const alice = await TestDataGenerator.generateDidKeyPersona();
+
+      const { message, dataBytes } = await TestDataGenerator.generateRecordsWrite({
+        author       : alice,
+        protocol     : PermissionsProtocol.uri,
+        protocolPath : 'invalid/path',
+        data         : Encoder.stringToBytes(JSON.stringify({}))
+      });
+
+      try {
+        PermissionsProtocol.validateSchema(message, dataBytes!);
+        expect.fail('Expected to throw');
+      } catch (error:any) {
+        expect(error.message).to.include(DwnErrorCode.PermissionsProtocolValidateSchemaUnexpectedRecord);
+        expect(error.message).to.include('invalid/path');
+      }
+    });
+
+    describe('validateScope', async () => {
```

Comment:
My main worry here is that the original feature tests are replaced with just unit tests. We don't actually know if the feature really works or not because there is no proof that `validateScope` is called.

At minimum as a sanity:
- Add a spy on `validateScope` in one of the permission tests above, probably the happy path, and make sure it is called.

Reply:
Doh! I actually noticed this while working one of the subsequent PRs and was sure i back ported it to this one. Will fix.